src/include/detail/graph/best_first.h

```diff
@@ -80,17 +78,12 @@ auto best_first_O0(
         continue;
 
       score_type heuristic = distance(db[neighbor_id], query);
-      if (heuristic == 0) {
-        std::cout << "Found goal!" << std::endl;
-        return true;
-      }
       pq.push({heuristic, neighbor_id});
       visited.insert(neighbor_id);
     }
   }
 
-  std::cout << "Goal not found: visited " << size(visited) << std::endl;
-  return false;
+  return visited;
```

Comment:
Can we avoid using `auto` return types in these functions? It is really confusing to change the return type of this function from `bool` to `set` and not understand if this will affect other parts of the code.

Reply:
Sure, done here. Prefer to leave the others b/c they turn out to be very, very long, and my original plan for all of this was to just modify how we set up these templates: instead of `template <class Graph,`, we could have `template <typename id_type` and then specify the `Graph` data structure explicitly. This would make the return type much shorter and more understandable.